remote_attestation/rust/src/handshaker.rs

```diff
@@ -21,203 +21,153 @@
 // - [`Server`] -> [`Client`]: [`ServerIdentity`]
 // - [`Client`] -> [`Server`]: [`ClientIdentity`]
 //
-// After the protocol handshake both sides create [`AeadEncryptor`] for exchanging encrypted
-// messages.
+// After the protocol handshake both sides create [`Encryptor`] for exchanging encrypted messages.
 
 use crate::{
     crypto::{
         get_random, get_sha256, AeadEncryptor, KeyNegotiator, KeyNegotiatorType, SignatureVerifier,
-        Signer, KEY_AGREEMENT_ALGORITHM_KEY_LENGTH, SIGNING_ALGORITHM_KEY_LENGTH,
+        Signer, SIGNING_ALGORITHM_KEY_LENGTH,
     },
     message::{
-        ClientHello, ClientIdentity, Deserializable, Serializable, ServerIdentity,
-        PROTOCOL_VERSION, REPLAY_PROTECTION_ARRAY_LENGTH,
+        deserialize, ClientHello, ClientIdentity, MessageWrapper, Serializable, ServerIdentity,
     },
     proto::{AttestationInfo, AttestationReport},
 };
 use anyhow::{anyhow, Context};
 use prost::Message;
 
-/// Client of the Remote attestation protocol handshake.
-pub struct ClientAttestationEngine<S>
-where
-    S: AttestationState,
-{
-    /// Behavior of the remote attestation protocol.
-    behavior: AttestationBehavior,
-    /// Current state of the remote attestation protocol.
-    state: S,
-    /// Collection of previously sent and received messages.
-    /// Signed transcript is sent in messages to prevent replay attacks.
-    transcript: Transcript,
+enum HandshakerState {
```

Comment:
Perhaps split this into client and server handshaker states. In that case, the client handshaker would better resemble the Java version.

Reply:
Done